test/hypothesis_utils.py

```diff
@@ -15,79 +16,107 @@
 # The tuples are (torch_quantized_dtype, zero_point_enforce), where the last
 # element is enforced zero_point. If None, any zero_point point within the
 # range of the data type is OK.
-ALL_QINT_TYPES = (
-    (torch.quint8, None),
-    (torch.qint8, None),
-    (torch.qint32, 0),  # We enforce zero_point = 0 for this case.
+
+# Tuple with all quantized data types.
+_ALL_QINT_TYPES = (
+    torch.quint8,
+    torch.qint8,
+    torch.qint32,
 )
 
-"""Strategy for generating test cases for quantized tensors.
-The resulting tensor is in float32 format.
+# Enforced zero point for every quantized data type.
+# If None, any zero_point point within the range of the data type is OK.
+_ENFORCED_ZERO_POINT = defaultdict(lambda: None, {
+    torch.quint8: None,
+    torch.qint8: None,
+    torch.qint32: 0
+})
+
+def _get_valid_min_max(qparams):
+    scale, zero_point, quantized_type = qparams
+    adjustment = 1 + torch.finfo(torch.float).eps
+
+    _long_type_info = torch.iinfo(torch.long)
+    long_min, long_max = _long_type_info.min / adjustment, _long_type_info.max / adjustment
+    # make sure intermediate results are within the range of long
+    min_value = max((long_min - zero_point) * scale, (long_min / scale + zero_point))
+    max_value = min((long_max - zero_point) * scale, (long_max / scale + zero_point))
+
+    return min_value, max_value
+
+"""Hypothesis filter to avoid overflows with quantized tensors.
 
 Args:
-    shapes: A list of shapes to generate.
-    dtypes: A list of data types to generate. See note below.
-    float_min, float_max: Min and max FP value for the output.
+    tensor: Tensor of floats to filter
+    qparams: Quantization parameters as returned by the `qparams`.
+
+Returns:
+    True
+
+Raises:
+    hypothesis.UnsatisfiedAssumption
+
+Note: This filter is slow. Use it only when filtering of the test cases is
```

Comment:
is this still slow?

Reply:
I didn't test if after the modification, but I would think so -- assume statements are usually slow :/

Reply:
Actually it is slow when tensors are large. Not because of the `min` or `max` but because it dumps the generated examples, instead of doing "constrained generation"